### PR TITLE
feat: added watch later page

### PR DESCRIPTION
**Pull request by Shubhamkr2610**
image

image

### CODE REVIEW

File: src/components/videoplayer/VideoPlayer.jsx [hunk 1]
```diff
@@ -16,6 +17,7 @@ export const VideoPlayer = () => {
     const {videoListId} = useParams()
     const {fetchLike , likeVideo} = useLike() 
     const {user} = useAuth();
+    const {postWatchLater} = useWatchLaterVideo()
     // console.log(user.encodedToken);
```
Comment (zoyatanweer) [inline src/components/videoplayer/VideoPlayer.jsx]: remove comment if not required

File: src/context/watchLaterContext.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import { toast } from 'react-toastify';
+
+const watchLaterContext = createContext();
+
+const WatchProvider= ({children})=>{
+    
+
+    const { user } = useAuth();
+    const { videoList } = useVideoList();
+    // const { videoListId } = useParams();
```
Comment (zoyatanweer) [inline src/context/watchLaterContext.js]: remove comment if not needed

File: src/context/watchLaterContext.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    const [ getWatchVideo, setGetWatchVideo] = useState([]);
+    const fetchWatchLater = async () => {
+      try {
+        const res = await axios.get("/api/user/watchlater", {
+          headers: {
+            authorization: user.encodedToken,
+          },
+        });
+        console.log(res.data);
```
Comment (zoyatanweer) [inline src/context/watchLaterContext.js]: remove console.log

File: src/context/watchLaterContext.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        const res = await axios.post(
+          "/api/user/watchlater",
+          { video: isVideoCardExist },
+          {
+            headers: {
+              authorization: token,
+            },
+          }
+        );
+        // console.log(res.data);
```
Comment (zoyatanweer) [inline src/context/watchLaterContext.js]: remove this

**Review by zoyatanweer — APPROVED** [review]
- Overall code looks good. Have given few suggestions so check them out. 
- In the UI, when video is added to watch later and removed, the toast displays "removed from likes". Change it into "remove from watch later".
- Great UI
- Good to merge

File: src/components/videoplayer/VideoPlayer.jsx (hunk 1 above)
Comment (Shubhamkr2610) [reply src/components/videoplayer/VideoPlayer.jsx]: removed

File: src/context/watchLaterContext.js (hunk 2 above)
Comment (Shubhamkr2610) [reply src/context/watchLaterContext.js]: removed

File: src/context/watchLaterContext.js (hunk 3 above)
Comment (Shubhamkr2610) [reply src/context/watchLaterContext.js]: removed

File: src/context/watchLaterContext.js (hunk 4 above)
Comment (Shubhamkr2610) [reply src/context/watchLaterContext.js]: removed, ThankYou

**Review by srishtigautam21 — APPROVED** [review]
The code clean
Good to merge

**Pull request merged**